diff -Nur avidemux-0.9pre5.orig/avidemux/ADM_audiofilter/audioeng_lame.cpp avidemux-0.9pre5/avidemux/ADM_audiofilter/audioeng_lame.cpp
--- avidemux-0.9pre5.orig/avidemux/ADM_audiofilter/audioeng_lame.cpp	Sat Jun  1 14:24:49 2002
+++ avidemux-0.9pre5/avidemux/ADM_audiofilter/audioeng_lame.cpp	Thu Jun 13 15:31:17 2002
@@ -17,6 +17,7 @@
  *                                                                         *
  ***************************************************************************/
 
+#include "lame/lame.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -34,7 +35,6 @@
 #include "audioprocess.hxx"
 #include "../toolkit.hxx"
 
-#include "lame/lame.h"
 lame_global_flags *myflags = NULL;
 
 // Ctor: Duplicate
