--- avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/multiplexor.cpp~	2008-01-01 18:38:25.000000000 +0100
+++ avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/multiplexor.cpp	2008-06-29 11:16:55.765736540 +0200
@@ -22,6 +22,7 @@
 #include <config.h>
 #include <math.h>
 #include <stdlib.h>
+#include <string.h>
 
 #include <mjpeg_types.h>
 #include <mjpeg_logging.h>
--- avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/inputstrm.cpp~	2007-08-03 02:04:40.000000000 +0200
+++ avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/inputstrm.cpp	2008-06-29 11:19:10.161766143 +0200
@@ -23,6 +23,7 @@
 
 #include <config.h>
 #include <assert.h>
+#include <limits.h>
 
 #include "mjpeg_types.h"
 #include "inputstrm.hpp"
--- avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/padstrm.cpp~	2007-08-03 02:04:40.000000000 +0200
+++ avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/padstrm.cpp	2008-06-29 11:19:49.865094597 +0200
@@ -23,6 +23,7 @@
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
+#include <string.h>
 
 #include "padstrm.hpp"
 
--- avidemux_2.4.1/avidemux/ADM_videoFilter/ADM_guiResize.cpp~	2007-09-25 21:01:54.000000000 +0200
+++ avidemux_2.4.1/avidemux/ADM_videoFilter/ADM_guiResize.cpp	2008-06-29 11:21:10.131913319 +0200
@@ -41,7 +41,7 @@
 #include <ADM_assert.h>
 
 
-static int getResizeParams(uint32_t * w, uint32_t * h, uint32_t * algo,uint32_t w,uint32_t h,uint32_t fps);
+static int getResizeParams(uint32_t * w, uint32_t * h, uint32_t * algo,uint32_t w2,uint32_t h2,uint32_t fps);
 
 
 extern uint8_t DIA_resize(uint32_t *width,uint32_t *height,uint32_t *algo,uint32_t originalw, uint32_t originalh,uint32_t fps1000);
--- avidemux_2.4.1/avidemux/ADM_script/DirectorySearch.h~	2007-06-09 16:38:45.000000000 +0200
+++ avidemux_2.4.1/avidemux/ADM_script/DirectorySearch.h	2008-06-29 11:24:26.278608518 +0200
@@ -22,6 +22,7 @@
 #include <sys/stat.h>
 #include <dirent.h>
 #endif
+#include <string.h>
 #include "StdFile.h"
 
 // create a non file bit for unix
--- avidemux_2.4.1/avidemux/gui_action.cpp~	2007-10-22 17:35:19.000000000 +0200
+++ avidemux_2.4.1/avidemux/gui_action.cpp	2008-06-29 11:25:49.725058609 +0200
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <string.h>
 #include <vector>
 using std::vector;
 #include <utility>
