--- avidemux_2.7.0/cmake/admFFmpegBuild_helpers.cmake.orig	2017-12-25 10:11:38.555601657 +0100
+++ avidemux_2.7.0/cmake/admFFmpegBuild_helpers.cmake	2017-12-25 14:01:39.232110743 +0100
@@ -304,7 +304,7 @@
        SET(PARRALLEL "-j4") 
     endif(NOT MSVC)
     add_custom_target(         libavutil_dummy ALL
-                                       COMMAND ${CMAKE_BUILD_TOOL}  ${PARRALLEL} # We assume make or gnumake when host is unix
+                                       COMMAND ${CMAKE_BUILD_TOOL} "V=1" # We assume make or gnumake when host is unix
                                        WORKING_DIRECTORY "${FFMPEG_BINARY_DIR}"
                                        COMMENT "Compiling FFmpeg")
 ENDMACRO(ADM_FF_ADD_DUMMY_TARGET)
