diff -urN avidemux-2.1.2.org/admin/detect-autoconf.sh avidemux-2.1.2/admin/detect-autoconf.sh
--- avidemux-2.1.2.org/admin/detect-autoconf.sh	2005-07-02 14:41:15.000000000 +0200
+++ avidemux-2.1.2/admin/detect-autoconf.sh	2007-01-04 17:16:16.539072500 +0100
@@ -21,16 +21,16 @@
     AUTOCONF="`$WHICH autoconf253`"
   elif test -x "`$WHICH autoconf-2.52`" ; then
     AUTOCONF="`$WHICH autoconf-2.52`"
-  elif test -x "`$WHICH autoconf2.50`" ; then
-    AUTOCONF="`$WHICH autoconf2.50`"
+  elif test -x "`$WHICH autoconf`" ; then
+    AUTOCONF="`$WHICH autoconf`"
   fi
 }
 
 checkAutoheader()
 {
-  if test -x "`$WHICH autoheader-2.5x`" ; then
-    AUTOHEADER="`$WHICH autoheader-2.5x`"
-    AUTOM4TE="`$WHICH autom4te-2.5x`"
+  if test -x "`$WHICH autoheader`" ; then
+    AUTOHEADER="`$WHICH autoheader`"
+    AUTOM4TE="`$WHICH autom4te`"
   elif test -x "`$WHICH autoheader-2.53`" ; then
     AUTOHEADER="`$WHICH autoheader-2.53`"
     AUTOM4TE="`$WHICH autom4te-2.53`"
@@ -47,9 +47,9 @@
 checkAutomakeAclocal ()
 {
   if test -z "$UNSERMAKE"; then
-    if test -x "`$WHICH automake-1.8`" ; then
-      AUTOMAKE="`$WHICH automake-1.8`"
-      ACLOCAL="`$WHICH aclocal-1.8`"
+    if test -x "`$WHICH automake`" ; then
+      AUTOMAKE="`$WHICH automake`"
+      ACLOCAL="`$WHICH aclocal`"
     elif test -x "`$WHICH automake-1.7`" ; then
       AUTOMAKE="`$WHICH automake-1.7`"
       ACLOCAL="`$WHICH aclocal-1.7`"
diff -urN avidemux-2.1.2.org/admin/detect-autoconf.sh.orig avidemux-2.1.2/admin/detect-autoconf.sh.orig
diff -urN avidemux-2.1.2.org/configure.in avidemux-2.1.2/configure.in
--- avidemux-2.1.2.org/configure.in.in	2006-03-01 22:33:35.000000000 +0100
+++ avidemux-2.1.2/configure.in.in	2007-01-04 17:20:59.936783750 +0100
@@ -37,8 +37,10 @@
 
 dnl Perform program name transformation
 AC_ARG_PROGRAM
+AC_GNU_SOURCE
 dnl Automake doc recommends to do this only here. (Janos)
 AM_INIT_AUTOMAKE(avidemux,"2.4.1")
+AM_PROG_AS
 ADM_SUBVERSION="0"
 
 if [[ -f .svn/entries ]]; then
@@ -79,7 +83,7 @@
 
 
 dnl _________________ gettext _________________
-AM_GNU_GETTEXT([external])
+AM_GNU_GETTEXT
 GETTEXT_PACKAGE=avidemux
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE], "$GETTEXT_PACKAGE", [gettext package name])
 ADM_LOCALEDIR=${datadir}/locale
