--- avidemux_2.7.8/cmake/admFFmpegBuild_helpers.cmake.orig	2021-07-04 06:51:39.736302112 +0200
+++ avidemux_2.7.8/cmake/admFFmpegBuild_helpers.cmake	2021-07-04 06:52:57.587005044 +0200
@@ -324,7 +324,7 @@ MACRO(ADM_FF_ADD_DUMMY_TARGET)
        endif(NOT NPROC EQUAL 0)
     endif(NOT MSVC)
     add_custom_target(         libavutil_dummy ALL
-                                       COMMAND ${CMAKE_BUILD_TOOL}  ${PARRALLEL} # We assume make or gnumake when host is unix
+                                       COMMAND ${CMAKE_BUILD_TOOL} "V=1" # We assume make or gnumake when host is unix
                                        WORKING_DIRECTORY "${FFMPEG_BINARY_DIR}"
                                        COMMENT "Compiling FFmpeg")
 ENDMACRO(ADM_FF_ADD_DUMMY_TARGET)
