--- avidemux_2.6.10/cmake/admFFmpegBuild_helpers.cmake.orig	2015-10-19 20:07:00.923333350 +0200
+++ avidemux_2.6.10/cmake/admFFmpegBuild_helpers.cmake	2015-10-19 20:34:06.114456242 +0200
@@ -259,7 +259,7 @@
 
     MESSAGE(STATUS "Configuring done, processing")
 
-    if (ADM_CPU_X86)
+    if (ADM_CPU_X86_X32 OR (ADM_CPU_X86_64 AND ADM_CPU_64BIT))
         file(READ ${FFMPEG_BINARY_DIR}/config.h FF_CONFIG_H)
         string(REGEX MATCH "#define[ ]+HAVE_YASM[ ]+1" FF_YASM "${FF_CONFIG_H}")
 
--- avidemux_2.6.10/cmake/admInstallDir.cmake.orig	2015-10-19 20:34:26.862201138 +0200
+++ avidemux_2.6.10/cmake/admInstallDir.cmake	2015-10-19 21:15:45.397657233 +0200
@@ -20,6 +20,8 @@
                 SET(AVIDEMUX_RELATIVE_LIB_DIR lib64)
         ELSEIF(CMAKE_C_IMPLICIT_LINK_DIRECTORIES MATCHES "\\/lib32([; ]|$)")
                 SET(AVIDEMUX_RELATIVE_LIB_DIR lib32)
+        ELSEIF(CMAKE_C_IMPLICIT_LINK_DIRECTORIES MATCHES "\\/libx32([; ]|$)")
+                SET(AVIDEMUX_RELATIVE_LIB_DIR libx32)
         ELSE(CMAKE_C_IMPLICIT_LINK_DIRECTORIES MATCHES "\\/lib64([; ]|$)")
                 SET(AVIDEMUX_RELATIVE_LIB_DIR lib)
         ENDIF(CMAKE_C_IMPLICIT_LINK_DIRECTORIES MATCHES "\\/lib64([; ]|$)")
--- avidemux_2.7.8/cmake/admYasm_gcc.cmake.orig	2021-07-03 20:36:14.136607549 +0200
+++ avidemux_2.7.8/cmake/admYasm_gcc.cmake	2021-07-03 20:40:46.251800041 +0200
@@ -34,7 +34,11 @@ IF(ADM_CPU_X86 )
       
      
     IF( ADM_CPU_X86_64 )
-        SET(ASM_ARGS_FORMAT ${ASM_ARGS_FORMAT} -m amd64 -DARCH_X86_64=1 -DARCH_X86_32=0)
+        IF( ADM_CPU_64BIT )
+            SET(ASM_ARGS_FORMAT "${ASM_ARGS_FORMAT} -m amd64 -DARCH_X86_64=1 -DARCH_X86_32=0")
+        ELSE( ADM_CPU_64BIT )
+            SET(ASM_ARGS_FORMAT "${ASM_ARGS_FORMAT} -m x32 -DARCH_X86_64=1 -DARCH_X86_32=0")
+        ENDIF( ADM_CPU_64BIT )
     ELSE( ADM_CPU_X86_64 )
         SET(ASM_ARGS_FORMAT ${ASM_ARGS_FORMAT}  -DARCH_X86_64=0 -DARCH_X86_32=1)
     ENDIF( ADM_CPU_X86_64 )
