--- avidemux_2.5.2/avidemux/ADM_coreImage/src/ADM_vidFieldASM.cpp~	2010-04-29 20:25:15.862622029 +0200
+++ avidemux_2.5.2/avidemux/ADM_coreImage/src/ADM_vidFieldASM.cpp	2010-04-29 21:30:35.374874733 +0200
@@ -25,7 +25,7 @@
 
 #include"ADM_vidField.h"
 
-#ifdef ADM_CPU_X86
+#ifdef ADM_CPU_X86_64
 //	#define DEBUG_DEINT 1
 //	#define MMX_TRACE 1
 	#include "ADM_mmxMacros.h"
@@ -81,7 +81,7 @@
                       }
            }
 }
-#ifdef ADM_CPU_X86
+#ifdef ADM_CPU_X86_64
 void ADMVideoFields::hasMotion_MMX(uint8_t *p,uint8_t *c,
 									uint8_t *n,
 									uint8_t *e,
--- avidemux_2.5.2/avidemux/ADM_coreImage/src/ADM_vidFields.cpp~	2010-04-29 20:25:15.871621667 +0200
+++ avidemux_2.5.2/avidemux/ADM_coreImage/src/ADM_vidFields.cpp	2010-04-29 21:31:24.857871769 +0200
@@ -103,7 +103,7 @@
            e2=_motionmask2+w; 	
   //___________________ C version of motion detection ________________________
        // other line
-#if defined(ADM_CPU_X86) && defined(ASM_DEINT)
+#if defined(ADM_CPU_X86_64) && defined(ASM_DEINT)
        if(CpuCaps::hasMMX())  
       	hasMotion_MMX(p,c,n,e,e2);
        else
