--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_license.cpp~	2009-12-16 13:05:49.000000000 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_license.cpp	2009-12-16 13:05:51.050534943 +0200
@@ -17,6 +17,7 @@
 
 #include "Q_license.h"
 #include "ADM_toolkitQt.h"
+#include <stdint.h>
 
 Ui_licenseWindow::Ui_licenseWindow(QWidget *parent) : QDialog(parent)
 {
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_jobs.cpp~	2009-12-16 12:45:16.000000000 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_jobs.cpp	2009-12-16 13:07:30.231566738 +0200
@@ -6,6 +6,7 @@
 #include "Q_jobs.h"
 #include "DIA_coreToolkit.h"
 #include "ADM_toolkitQt.h"
+#include <stdio.h>
 
 static void updateStatus(void);
 extern bool parseECMAScript(const char *name);
--- avidemux_2.5.0/avidemux/ADM_coreUI/include/DIA_coreToolkit.h~	2009-12-16 12:45:15.000000000 +0200
+++ avidemux_2.5.0/avidemux/ADM_coreUI/include/DIA_coreToolkit.h	2009-12-16 13:09:16.437182359 +0200
@@ -19,6 +19,8 @@
 #ifndef ADM_COREUI_TOOLKIT__
 #define ADM_COREUI_TOOLKIT__
 
+#include <stdint.h>
+
 typedef enum 
 {
         ADM_LOG_NONE=0,
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.h~	2009-12-16 12:45:16.000000000 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.h	2009-12-16 13:30:37.550386331 +0200
@@ -1,6 +1,8 @@
 #ifndef ADM_Q_SLIDER_H
 #define ADM_Q_SLIDER_H
 
+#include <stdint.h>
+
 class ADM_QSlider : public QSlider
 {
 protected:
