--- avidemux_2.6.9/avidemux/gtk/CMakeLists.txt~	2015-05-24 11:48:10.000000000 +0200
+++ avidemux_2.6.9/avidemux/gtk/CMakeLists.txt	2015-05-24 11:56:38.844485490 +0200
@@ -33,7 +33,7 @@
 ADD_SUBDIRECTORY(ADM_userInterfaces ./ADM_userInterfacesGtk)
 
 SDLify(../common/main.cpp)
-ADD_SOURCE_CFLAGS(../common/main.cpp "-DADM_SUBVERSION=\"${ADM_SUBVERSION}\"")
+ADD_SOURCE_CFLAGS(../common/main.cpp "-DADM_SUBVERSION=\"${ADM_SUBVERSION}\""  "-DADM_VERSION='\"${AVIDEMUX_VERSION}\"'")
 
 ###########################################
 # Version Info
--- avidemux_2.6.11/avidemux/gtk/ADM_userInterfaces/ADM_gui2/GUI_gtkRender.cpp.orig	2016-01-30 08:54:43.917653247 +0100
+++ avidemux_2.6.11/avidemux/gtk/ADM_userInterfaces/ADM_gui2/GUI_gtkRender.cpp	2016-01-30 10:22:47.504098179 +0100
@@ -110,9 +110,9 @@
 		xinfo->display = (void*)GDK_WINDOW_HWND(win);
 #elif defined(__APPLE__)
 		xinfo->display = 0;
-		xinfo->window = getMainNSWindow();
+		xinfo->systemWindowId = getMainNSWindow();
 #else
-		xinfo->window = GDK_WINDOW_XID(win);
+		xinfo->systemWindowId = GDK_WINDOW_XID(win);
 		xinfo->display = GDK_WINDOW_XDISPLAY(parentWin);
 #endif
 
--- avidemux_2.6.11/avidemux/gtk/ADM_userInterfaces/ADM_gui2/GUI_bindings.cpp.orig	2016-01-30 10:29:43.714080713 +0100
+++ avidemux_2.6.11/avidemux/gtk/ADM_userInterfaces/ADM_gui2/GUI_bindings.cpp	2016-01-30 19:32:06.962714994 +0100
@@ -1387,6 +1387,19 @@
     };
 
 /**
+ * \fn UI_reset
+ * \brief reset
+ * @return 
+ */
+bool  	UI_reset(void)
+{
+    UI_setVideoCodec(0);
+    UI_setAudioCodec(0);
+    UI_setCurrentPreview(false);
+    return true;
+}
+
+/**
  *      \fn UI_Init
  *      \brief Entry point. Initialize renderLib.
  *
@@ -1596,6 +1596,15 @@
     return true;
 }
 /**
+ * \fn UI_setDisplayName
+ * \brief display current displayEngine name
+ */
+bool UI_setDisplayName(const char *name)
+{
+    // TODO?
+    return true;
+}
+/**
 	\fn UI_setTimeShift
 
 */
